.azure-pipelines/ultimate-pipeline.yml

```
@@ -4989,107 +4989,163 @@ stages:
-          SCENARIO: "+S TRACE_PROPAGATION_STYLE_W3C +S PROFILING +S LIBRARY_CONF_CUSTOM_HEADERS_SHORT +S LIBRARY_CONF_CUSTOM_HEADERS_LONG"
-        remainder_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: "+S TRACE_PROPAGATION_STYLE_W3C +S PROFILING +S LIBRARY_CONF_CUSTOM_HEADERS_SHORT +S LIBRARY_CONF_CUSTOM_HEADERS_LONG"
-        debugger_poc:
-          WEBLOG_VARIANT: "poc"
-          SCENARIO: DEBUGGER_SCENARIOS
-        debugger_uds:
-          WEBLOG_VARIANT: "uds"
-          SCENARIO: DEBUGGER_SCENARIOS
+        - script: |
+            PACKAGE_NAME=$(basename $(Build.ArtifactStagingDirectory)/datadog-dotnet-apm-*.tar.gz)
+            echo Moving $PACKAGE_NAME to system-tests/binaries
+            mv $(Build.ArtifactStagingDirectory)/$PACKAGE_NAME system-tests/binaries/
+          displayName: Move dotnet binary to system test folder
+          condition: eq( variables['System.JobAttempt'], 1 )
+        
+        - script: |
+            set -e
+            cd system-tests
+            mkdir -p ../docker-images
+        
+            while read w; do
+              echo "------------------------------------"
+              echo "Building images for variant: $w"
+              echo "------------------------------------"
+              # Build the images. For example:
+              ./build.sh dotnet --weblog-variant $w
+            
+              # Save weblog variant image
+              docker save system_tests/weblog:latest -o ../docker-images/$w.tar
+            done < /tmp/weblogs_list.txt
+            
+            # Save agent image
+            docker save system_tests/agent:latest -o ../docker-images/agent.tar
 
-    steps:
+          displayName: Build weblogs and agent & save Docker images
+          condition: eq( variables['System.JobAttempt'], 1 )
+        
+        - task: PublishPipelineArtifact@1
+          displayName: Publish Docker images artifact
+          condition: eq( variables['System.JobAttempt'], 1 )
+          inputs:
+            path: 'docker-images'
+            artifactName: 'docker-images'
+
+    - job: tests
+      displayName: ""
+      timeoutInMinutes: 60
+      pool:
+          vmImage: ubuntu-latest
+      dependsOn: compute_scenarios
+     
+      strategy:
+        matrix: $[ dependencies.compute_scenarios.outputs['create_matrix.matrixJson'] ]
+
+      steps:
       - checkout: none
 
       - task: UsePythonVersion@0
         inputs:
-            versionSpec: '3.12'
+          versionSpec: '3.12'
         displayName: Install python 3.12
 
       - script: git clone --depth 1 https://github.com/DataDog/system-tests.git
         displayName: Get system tests repo
-
+        
       - task: DownloadPipelineArtifact@2
-        displayName: Download linux-packages
+        displayName: Download Docker images
         inputs:
-          artifact: linux-packages-linux-x64
-          patterns: '**/*tar.gz'
-          path: $(Build.ArtifactStagingDirectory)
+          artifact: docker-images
```

If you update the above you need this:
```suggestion
          artifact: system-test-docker-images
```